flask/app.py

```diff
@@ -1029,6 +1033,32 @@ def index():
         rule = self.url_rule_class(rule, methods=methods, **options)
         rule.provide_automatic_options = provide_automatic_options
 
+        # a bound subdomain's null value is '' and a unbound is None.
+        current_subdomain = rule.subdomain
+        if current_subdomain is None:
+            # Normalizing.
+            current_subdomain = ''
+        prior_rule_check = (rule.rule, current_subdomain)
+        if prior_rule_check in self.prior_rules:
+            # static/<path:filename> is added once per blueprint. Ignoring for
+            # this check.
+            if rule.rule != "/static/<path:filename>":
+                prior_rule_methods = self.prior_rules.get(
+                    prior_rule_check,
+                    None)
+                # Flask adds non-specified methods to routes. Removing the
```

Comment:
Flask adds a `OPTIONS` method to every URL (that's the `required_methods` variable). I might need to add a unit test to make sure that options methods are still routable.